crates/fuel-core/src/query/message.rs

```diff
@@ -166,61 +165,57 @@ pub fn message_proof<T: MessageProofData + ?Sized>(
             _ => None,
         });
 
-    let (sender, recipient, nonce, amount, data) = match receipt {
-        Some(r) => r,
-        None => return Ok(None),
+    let Some((sender, recipient, nonce, amount, data)) = receipt else {
+        return Err(
+            anyhow::anyhow!("desired `nonce` missing in transaction receipts").into(),
+        )
+    };
+
+    let Some(data) = data else {
+        return Err(anyhow::anyhow!("output message doesn't contain any `data`").into())
     };
-    let data =
-        data.ok_or(anyhow::anyhow!("Output message doesn't contain any `data`"))?;
 
     // Get the block id from the transaction status if it's ready.
-    let message_block_height = match database
+    let Some(TransactionStatus::Success { block_height, .. }) = database
         .transaction_status(&transaction_id)
-        .into_api_result::<TransactionStatus, StorageError>(
-    )? {
-        Some(TransactionStatus::Success { block_height, .. }) => block_height,
-        _ => return Ok(None),
+        .into_api_result::<TransactionStatus, StorageError>()?
+    else {
+        return Err(anyhow::anyhow!("unable to obtain block height").into())
     };
 
     // Get the message fuel block header.
-    let (message_block_header, message_block_txs) = match database
-        .block(&message_block_height)
+    let Some(block) = database
+        .block(&block_height)
         .into_api_result::<CompressedBlock, StorageError>()?
```

Comment:
I like old naming for vairables more=)

Reply:
`message_block` and `message_block_height` are back in https://github.com/FuelLabs/fuel-core/pull/2258/commits/0c2dbc5743634bad1c5a67aed2765ce6fe9e731c